framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -276,7 +276,18 @@ public class BaseTypeVisitor<Factory extends GenericAnnotatedTypeFactory<?, ?, ?
   /** The tree of the enclosing method that is currently being visited, if any. */
   protected @Nullable MethodTree methodTree = null;
 
+  /** The number of seconds that typechecking must take, to issue a "slow typechecking" warning. */
+  protected long slowTypecheckingSeconds = 30;
```

Comment:
Just a suggestion:
I think that people might not only want to use this functionality for this specific issue but maybe some people would like to use it to debug slow builds in general (for example if a build has 20 occurrences of code each taking 20 seconds to typecheck, the people might want to know where these are).
For this, it might be useful to configure this duration to find parts they should change.